plugins/scaffolder-backend/src/scaffolder/stages/publish/publishers.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TemplateEntityV1alpha1 } from '@backstage/catalog-model';
+import { parseLocationAnnotation } from '../helpers';
+import { PublisherBase, PublisherBuilder } from './types';
+import { RemoteProtocol } from '../types';
+
+export class Publishers implements PublisherBuilder {
+  private publisherMap = new Map<RemoteProtocol, PublisherBase>();
+
+  register(protocol: RemoteProtocol, publisher: PublisherBase) {
+    this.publisherMap.set(protocol, publisher);
+  }
+
+  get(template: TemplateEntityV1alpha1): PublisherBase {
```

Comment:
I'm not sure if it makes sense to derive the publisher from the entity. A template can absolutely be stored in, e.g., public GitHub and the newly created repo should be pushed to a private GitLab instance. It'd probably be best to let the user choose if there's more than one publisher configured, but for now, I think it's probably a fair assumption that in most cases the target repo and the template will live in the same GitHub/GitLab instance.

Reply:
That is 100% the end goal. We ideally want to have a list of publishers to choose from, but we just haven't gotten to that stage yet.

Ideally we would be able to do `/v1/scaffolder/publishers` and display this list of publishers in the frontend when choosing the stages.

I will create a ticket for this.